pages-e2e/features/simplePagesSsrRoutes/navigation.test.ts

```diff
@@ -0,0 +1,36 @@
+import { describe, it } from 'vitest';
+import { getAssertVisible } from '@features-utils/getAssertVisible';
+import { collectHardNavigationRequests } from '@features-utils/collectHardNavigationRequests';
+
+describe('Simple Pages SSR routes SPA navigation', () => {
+	it('should soft navigate between static routes', async ({ expect }) => {
+		const page = await BROWSER.newPage();
+		const assertVisible = getAssertVisible(page);
+
+		const hardNavigationRequests = await collectHardNavigationRequests(page);
+
+		await page.goto(`${DEPLOYMENT_URL}/ssr-navigation`);
+
+		await assertVisible('h1', { hasText: 'Navigation' });
+
+		await assertVisible('h2', { hasText: 'Server Side Rendered Index' });
+
+		await page.locator('a', { hasText: 'to page A' }).click();
+
+		await assertVisible('h2', { hasText: 'Server Side Rendered Page A' });
+
+		await page.locator('a', { hasText: 'to page B' }).click();
+
+		await assertVisible('h2', { hasText: 'Server Side Rendered Page B' });
+
+		await page.goBack();
+
+		await assertVisible('h2', { hasText: 'Server Side Rendered Page A' });
+
+		// Note: we don't check the hardNavigationRequests here as this is broken in Next 12.2
+		//       we can either: fix for 12.2 (if worth it), find a way to discern and avoid these
+		//       checks if we are in 12.2 or uncomment the lines below when/if we drop support for 12
```

Comment:
I've looked into it and it's actually caused by the Vercel output not being properly produced, so I think we cannot actually fix this on our end, Next 12.3 fixes it so I am just going to bump the fixture's version

(so I'd imagine that we can say that 12.3 is the oldest version we fully support 🤔)

(PS: 12.2 still worked but here we would get full page reloads when they should not happen)